mlflow/pyfunc/__init__.py

```diff
@@ -1824,19 +2032,28 @@ def spark_udf(
 
     if (
         is_spark_connect
+        and not is_dbconnect_mode
         and env_manager in (_EnvManager.VIRTUALENV, _EnvManager.CONDA)
-        and not should_spark_connect_use_nfs
     ):
         raise MlflowException.invalid_parameter_value(
-            f"Environment manager {env_manager!r} is not supported in Spark connect mode "
-            "when either non-Databricks environment is in use or NFS is unavailable.",
+            f"Environment manager {env_manager!r} is not supported in Spark Connect "
+            "client environment if it connects to non-Databricks Spark cluster.",
         )
 
     local_model_path = _download_artifact_from_uri(
         artifact_uri=model_uri,
         output_path=_create_model_downloading_tmp_dir(should_use_nfs),
     )
 
+    if prebuilt_env_path:
+        _verify_prebuilt_env(spark, local_model_path, prebuilt_env_path)
+        env_manager = _EnvManager.VIRTUALENV
```

Comment:
why do we need to `env_manager` here?

Reply:
the prebuilt env is a virtualenv env, so the following code reuses code of _EnvManager.VIRTUALENV case

Reply:
I see, can we avoid mutating `env_manager` because it might surprise us in the future.

Reply:
I'll make `env_manager` argument default value to be `None` (means unset), for the unset case, then if `prebuilt_env_path` is set, then `env_manager` is set to `virtualenv`, otherwise `env_manager` is set to 'local'.
this way should make code clear.